### ISSUE TITLE
Add ability to re-run/resume post-processors

### ISSUE THREAD

**Issue by dingus9** [hashicorp/packer#9686]
## Feature
Allow the resuming of your build at the post-processors stage. It would be a nice feature to be able to have the output of the build stage remain in place until after a successful post-processor run.

## Reasoning: 
Sometimes there are errors in the post-processor section that are out of the control of the build stage. At the moment debugging post-processors seems to require the entire rebuilding of the artifact. Things that I've encountered so far that have failed in the post-processors have been authentication errors, an error where the `{{.Provider}}` wasn't being inserted into the vagrant-cloud post-processor and a network error which hung the build at the post-processor. All of these errors required me to re-run the entire build before I could reliably tell if it would pass that phase or not. It would be a huge benefit when debugging builds to be able to run the post-processor stage on a successful build, but failed post-processor without having to re-run the entire build.

**Comment by dingus9** [comment]
P.S. This would be kind of like the --provision flag on vagrant up

**Comment by nywilken** [comment]
Hi @dingus9 totally understand where you are coming from. This sounds like a good idea but not sure if this is something that we will be able to dive deeper into any time soon. 

Have you had a chance to take a look at the our recently added pipelineing guide https://www.packer.io/guides/packer-on-cicd/pipelineing-builds? It might be a good way forward to help simplify things.

**Comment by SwampDragons** [comment]
I agree this would be a big improvement; in the meantime using the null builder and artifice post-processor to run your post-processor flow can provide you with a nice workaround.

**Comment by yves-vogl** [comment]
+1 - when building the virtual-environments from GitHub Actions Repository you may end up with build times round about 1 hour.
